app.py

```diff
@@ -1,353 +1,90 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
 #
-#        http://www.apache.org/licenses/LICENSE-2.0
+#     http://www.apache.org/licenses/LICENSE-2.0
 #
-#    Unless required by applicable law or agreed to in writing, software
-#    distributed under the License is distributed on an "AS IS" BASIS,
-#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-#    See the License for the specific language governing permissions and
-#    limitations under the License.
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
 from __future__ import annotations
 
-from typing import TYPE_CHECKING, Any
+import argparse
 
+import dash
 import diskcache
-from dash import Dash, DiskcacheManager, Input, MATCH, Output, State, ctx, no_update
-from dash.exceptions import PreventUpdate
-
-import employee_scheduling
-import utils
-from app_configs import (APP_TITLE, DEBUG, LARGE_SCENARIO, MEDIUM_SCENARIO, MIN_MAX_EMPLOYEES,
-                         SMALL_SCENARIO)
-from app_html import errors_list, set_html
+from dash import DiskcacheManager
 
-if TYPE_CHECKING:
-    from pandas import DataFrame
+from demo_configs import APP_TITLE, THEME_COLOR, THEME_COLOR_SECONDARY
+from demo_interface import create_interface
 
-cache = diskcache.Cache("./cache")
-background_callback_manager = DiskcacheManager(cache)
+# Essential for initializing callbacks. Do not remove.
+import demo_callbacks
 
-# Fix for Dash background callbacks crashing on macOS 10.13+ (https://bugs.python.org/issue33725)
-# See https://github.com/dwave-examples/flow-shop-scheduling/pull/4 for more details.
+# Fix Dash long callbacks crashing on macOS 10.13+ (also potentially not working
+# on other POSIX systems), caused by https://bugs.python.org/issue33725
+# (aka "beware of multithreaded process forking").
+#
+# Note: default start method has already been changed to "spawn" on darwin in
+# the `multiprocessing` library, but its fork, `multiprocess` still hasn't caught up.
+# (see docs: https://docs.python.org/3/library/multiprocessing.html#contexts-and-start-methods)
 import multiprocess
+
 if multiprocess.get_start_method(allow_none=True) is None:
-    multiprocess.set_start_method('spawn')
+    multiprocess.set_start_method("spawn")
+
+cache = diskcache.Cache("./cache")
+background_callback_manager = DiskcacheManager(cache)
 
-app = Dash(
+app = dash.Dash(
     __name__,
     meta_tags=[{"name": "viewport", "content": "width=device-width, initial-scale=1"}],
     prevent_initial_callbacks="initial_duplicate",
     background_callback_manager=background_callback_manager,
 )
 app.title = APP_TITLE
 
+app.config.suppress_callback_exceptions = True
```

Comment:
Why is this needed?

Reply:
It's an aspect of the template that disabled Dash's default check of scanning every callback Input, Output, State, etc to make sure the Dash object associated with that id exists in the layout. We don't want this usually because there's cases when we add something to the layout and then have a callback associated with it and if this wasn't disabled it would error because that object doesn't exist on first load.